src/Common/src/Interop/Unix/libssl/SecuritySafeHandles.cs

```diff
@@ -265,7 +276,9 @@ public override bool IsInvalid
 
         protected override bool ReleaseHandle()
         {
-            Interop.OpenSsl.FreeSslContext(_sslContext);
+            Interop.OpenSsl.Disconnect(_sslContext);
+            _sslContext.DangerousRelease();
+            _sslContext.Dispose();
```

Comment:
>  It seems like you're still in the exact same situation as right now

The difference is that earlier SSL_free (via SafeSslContext.ReleaseHandle) was also closing the connection causing a later FreeSslContext to fail.

But yes, _sslContext.Dispose can be moved to a Dispose override